testing_utilities.go

```diff
@@ -251,8 +251,8 @@ func decimalStrictEquals(t *testing.T, decimal1, decimal2 *ion.Decimal) {
 func writeFromReaderToWriter(t *testing.T, reader ion.Reader, writer ion.Writer) {
 	for reader.Next() {
 		name := reader.FieldName()
-		if name != "" {
-			require.NoError(t, writer.FieldName(name), "Something went wrong executing writer.FieldName(name)")
+		if name != nil {
```

Comment:
I know it is an existing logic but are you sure it shouldn't be the other way round?
```if name == nil { something went wrong } ```

Reply:
It should be 

* `nil` something went wrong when getting the FieldName() check the reader context for errors 
* non-`nil` you got a fieldName proceed 

https://github.com/amzn/ion-go/pull/93#discussion_r456037974

Reply:
The "Something went wrong" part in the require() is referring to something going wrong during writer.FieldName(*name)

Reply:
I'll add an assert/require.NotNil() to handle this scenario.

Reply:
Addressed